gguf-py/scripts/gguf-dump.py

```diff
@@ -101,25 +101,285 @@ def dump_metadata_json(reader: GGUFReader, args: argparse.Namespace) -> None:
+        {'key_name':'field_name',       'header_name':'Key',      'align':'left'},
+        {'key_name':'value',            'header_name':'Value',    'align':'left'},
+    ]
+
+    markdown_content += markdown_table_with_alignment_support(kv_dump_table_header_map, kv_dump_table)
+
+    markdown_content += "\n"
+
+    if not args.no_tensors:
+        # Group tensors by their prefix and maintain order
+        tensor_prefix_order: list[str] = []
+        tensor_name_to_key: dict[str, int] = {}
+        tensor_groups: dict[str, list[ReaderTensor]] = {}
+        total_elements = sum(tensor.n_elements for tensor in reader.tensors)
+
+        # Parsing Tensors Record
+        for key, tensor in enumerate(reader.tensors):
+            tensor_components = tensor.name.split('.')
+
+            # Classify Tensor Group
+            tensor_group_name = "base"
+            if tensor_components[0] == 'blk':
+                tensor_group_name = f"{tensor_components[0]}.{tensor_components[1]}"
+
+            # Check if new Tensor Group
+            if tensor_group_name not in tensor_groups:
+                tensor_groups[tensor_group_name] = []
+                tensor_prefix_order.append(tensor_group_name)
+
+            # Record Tensor and Tensor Position
+            tensor_groups[tensor_group_name].append(tensor)
+            tensor_name_to_key[tensor.name] = key
+
+        # Tensors Mapping Dump
+        markdown_content += f'## Tensors Overview {element_count_rounded_notation(total_elements)} Elements\n\n'
+        markdown_content += f'Total number of elements in all tensors: {total_elements} Elements\n'
+        markdown_content += '\n'
+
+        for group in tensor_prefix_order:
+            tensors = tensor_groups[group]
+            group_elements = sum(tensor.n_elements for tensor in tensors)
+            markdown_content += f"- [{translate_tensor_name(group)} Tensor Group - {element_count_rounded_notation(group_elements)} Elements](#{group.replace('.', '_')})\n"
+
+        markdown_content += "\n"
+
+        for group in tensor_prefix_order:
+            tensors = tensor_groups[group]
+            group_elements = sum(tensor.n_elements for tensor in tensors)
+            group_percentage = group_elements / total_elements * 100
+            markdown_content += f"### <a name=\"{group.replace('.', '_')}\">{translate_tensor_name(group)} Tensor Group : {element_count_rounded_notation(group_elements)} Elements</a>\n\n"
+
+            # Precalculate column sizing for visual consistency
+            prettify_element_est_count_size: int = 1
+            prettify_element_count_size: int = 1
+            prettify_dimension_max_widths: dict[int, int] = {}
+            for tensor in tensors:
+                prettify_element_est_count_size = max(prettify_element_est_count_size, len(str(element_count_rounded_notation(tensor.n_elements))))
+                prettify_element_count_size = max(prettify_element_count_size, len(str(tensor.n_elements)))
```

Comment:
This is better, but I still feel like sub-column alignment isn't fully solved, although now the incorrect cases are extremely rare and won't really happen in practice.

For example, when the name of the column is wider than the content, this doesn't spread the empty space correctly. See the `Elements` column with a very small model:

```markdown
| T_ID | Tensor Layer Name        | Human Friendly Tensor Layer Name               | Elements | Shape           | Type |
|-----:|:-------------------------|:-----------------------------------------------|:---------|:----------------|:-----|
|    2 | blk.0.attn_norm.weight   | Block 0 Attention Normalization (W)            | ( 6)  6  |  6 x  1 x 1 x 1 | F32  |
|    3 | blk.0.ffn_down.weight    | Block 0 Feed-Forward Network "Down" (W)        | (96) 96  | 16 x  6 x 1 x 1 | F32  |
|    4 | blk.0.ffn_gate.weight    | Block 0 Feed-Forward Network "Gate" (W)        | (96) 96  |  6 x 16 x 1 x 1 | F32  |
|    5 | blk.0.ffn_up.weight      | Block 0 Feed-Forward Network "Up" (W)          | (96) 96  |  6 x 16 x 1 x 1 | F32  |
|    6 | blk.0.ffn_norm.weight    | Block 0 Feed-Forward Network Normalization (W) | ( 6)  6  |  6 x  1 x 1 x 1 | F32  |
|    7 | blk.0.attn_k.weight      | Block 0 Attention Key (W)                      | (12) 12  |  6 x  2 x 1 x 1 | F32  |
|    8 | blk.0.attn_output.weight | Block 0 Attention Output (W)                   | (36) 36  |  6 x  6 x 1 x 1 | F32  |
|    9 | blk.0.attn_q.weight      | Block 0 Attention Query (W)                    | (36) 36  |  6 x  6 x 1 x 1 | F32  |
|   10 | blk.0.attn_v.weight      | Block 0 Attention Value (W)                    | (12) 12  |  6 x  2 x 1 x 1 | F32  |
```

But this seems acceptable, so feel free to leave this as it is.